src/main/java/com/maddyhome/idea/vim/extension/searchindex/VimSearchIndexExtension.kt

```diff
@@ -0,0 +1,183 @@
+    VimExtensionFacade.executeNormalWithoutMapping(keys, editor.ij)
+    VimSearchIndexExtension.updateSearchIndex(editor.ij, VimPlugin.getSearch())
+  }
+}
+
+internal class VimSearchIndexExtension : VimExtension, ModeChangeListener {
+  init {
+    LOG.info("Looking at SearchGroup implementation: ${VimPlugin.getSearch().javaClass}")
+  }
+
+  companion object {
+    private val LOG = logger<VimSearchIndexExtension>()
+
+    fun updateSearchIndex(editor: Editor, searchGroup: IjVimSearchGroup) {
+      if (searchGroup.lastSearchPattern.isNullOrEmpty()) {
+        return
+      }
+      val pattern = searchGroup.lastSearchPattern
+      if (pattern.isNullOrEmpty()) {
+        return
+      }
+
+      LOG.info("search index - Processing pattern: $pattern")
+
+      // FIXME: global search option -> previous search option
+      val ignoreCase = injector.options(editor = editor.vim).ignorecase
+      LOG.info("Ignore case: $ignoreCase")
+
+      val results = injector.searchHelper.findAll(
+        editor = editor.vim,
+        pattern = pattern,
+        startLine = 0,
+        endLine = -1,
+        ignoreCase = ignoreCase,
+      )
+
+      val label = ExEntryPanel.getInstance().label
+      val (currentIndex: Int, resultSize: Int) = if (results.isNotEmpty()) {
+        Pair(first = getCurrentIndex(editor = editor, results = results), second = results.size)
+      } else {
+        Pair(first = 0, second = 0)
+      }
+
+      ApplicationManager.getApplication().invokeLater {
+        LOG.info("searchindex - [${currentIndex}/${resultSize}] $label$pattern")
+        SearchIndex.update("[${currentIndex}/${resultSize}] $label$pattern")
+      }
+    }
+
+    private fun getCurrentIndex(
+      editor: Editor,
+      results: List<TextRange>,
+    ): Int {
+      val caretOffset = editor.caretModel.offset
+      var current = 0
+      results.forEachIndexed { index, range ->
+        if (range.startOffset <= caretOffset && caretOffset <= range.endOffset) {
+          current = index + 1
+        }
+      }
+      return current
+    }
+  }
+
+  override fun getName(): String = "searchindex"
+
+  override fun init() {
+    LOG.info("Initializing SearchIndex extension")
+    injector.listenersNotifier.modeChangeListeners.add(this)
+
+    listOf("n", "N", "*", "#", "g*", "g#").forEach {
+      val key = injector.parser.parseKeys(it)
+      VimExtensionFacade.putExtensionHandlerMapping(
+        modes = setOf(MappingMode.NORMAL, MappingMode.VISUAL),
+        fromKeys = key,
+        pluginOwner = owner,
+        extensionHandler = SearchAgainHandler(key),
+        recursive = false,
+      )
+    }
+  }
+
+  override fun dispose() {
+    LOG.info("Disposing SearchIndex extension")
+    injector.listenersNotifier.modeChangeListeners.remove(this)
+  }
+
+  override fun modeChanged(editor: VimEditor, oldMode: Mode) {
+    LOG.info("Mode changed from $oldMode to ${editor.mode}")
+
+    if (oldMode is Mode.CMD_LINE && editor.mode !is Mode.CMD_LINE) {
```

Comment:
I ended up using modeChanged since I couldn't find a dedicated search event listener. If there's a proper search event that I missed, please let me know.